music21/interval.py

```diff
@@ -3232,6 +3259,10 @@ def isStep(self) -> bool:
     def isSkip(self) -> bool:
         return self.diatonic.isSkip
 
+    @property
+    def transpositionChangesPitchClass(self) -> bool:
+        return self.simpleName != 'P1'
```

Comment:
I've changed the property name to transpositionChangesPitchName, but left the "self.semitones is not fractional" check in place for now, since otherwise it gives wrong answers for P1 +/- half-a-semitone.